tracing/span.go

```diff
@@ -320,81 +425,116 @@ func (s *Span) CreateLocalChildForContext(ctx context.Context, name string, comp
 // Parent id will be inherited from the span id,
 // and span id will be randomly generated.
 // Trace id, sampled, and flags will be copied over.
-func (s *Span) CreateClientChild(name string) *Span {
-	return s.createChild(name, SpanTypeClient, "")
+func (s *baseSpan) CreateClientChild(name string) *ClientSpan {
+	child := &ClientSpan{
+		baseSpan: baseSpan{
+			trace: newTrace(s.trace.tracer, name),
+		},
+	}
+
+	child.trace.parentID = s.trace.spanID
+	child.trace.traceID = s.trace.traceID
+	child.trace.sampled = s.trace.sampled
+	child.trace.flags = s.trace.flags
+	return child
 }
 
-// CreateClientChildForContext creates a client child-span with given name and a
-// context that can be used by the client you are creating the span for.
+// CreateLocalChild creates a local child-span with given name and component.
 //
-// A client child-span should be used to make requests to other upstream
-// servers.
+// component is optional local component name and could be empty string.
 //
 // Timestamps, counters, and tags won't be inherited.
 // Parent id will be inherited from the span id,
 // and span id will be randomly generated.
 // Trace id, sampled, and flags will be copied over.
-func (s *Span) CreateClientChildForContext(ctx context.Context, name string) (context.Context, *Span) {
-	child := s.CreateClientChild(name)
-	return context.WithValue(ctx, childSpanKey, child), child
+func (s *baseSpan) CreateLocalChild(name, component string) *LocalSpan {
+	child := newLocalSpan(s.trace.tracer, name, component)
+
+	child.trace.parentID = s.trace.spanID
+	child.trace.traceID = s.trace.traceID
+	child.trace.sampled = s.trace.sampled
+	child.trace.flags = s.trace.flags
+	return child
 }
 
-func (s *Span) createChild(name string, spanType SpanType, component string) *Span {
-	span := newSpan(s.tracer, name, spanType)
-	span.spanType = spanType
-	span.component = component
+// ServerSpan defines a tracing span over an entire request to a server.
+type ServerSpan struct {
```

Comment:
It should be mentioned in doc comment that it implements Span. Same as the other exported concrete types.